crypto/src/client.rs

```diff
@@ -30,6 +30,7 @@ use openmls_traits::{
 
 pub(crate) const INITIAL_KEYING_MATERIAL_COUNT: usize = 100;
 
+/// Wrapper on the byte array used to identify a client id
```

Comment:
> Prefer functional description instead of technical one here i.e. explain what a client id is from Wire POV (unique client identifier, != userId) and also its format userID:clientID@domain and an example `b455a431-9db6-4404-86e7-6a3ebe73fcaf:3ae58155@mls.example.com`

Not sure we want this. This is only relevant in the context of Wire Cloud & CoreLogic interacting with each other. CoreCrypto intentionally performs no checks - it used to! (see here: https://github.com/wireapp/core-crypto/commit/7ba4635264c51c5900af656c420e59b66deae060#diff-35ce8a4538af9211db85cce68688f1c44d2aeeeaa1d0c3e834483ba7dec722e4L28)

But this was explicitely asked to be removed for CoreCrypto to treat any identifier as completely opaque, so we shouldn't document something that we never ever check